jenkins/L0_Test.groovy

```diff
@@ -210,6 +265,206 @@ def runLLMTestlistOnSlurm(pipeline, platform, testList, config=VANILLA_CONFIG, p
+    SlurmCluster cluster = SlurmConfig.clusterConfig[partition.clusterName]
+
+    def jobUID = "${cluster.host}-multi_node_test-${UUID.randomUUID().toString()}"
+
+    try {
+        // Run ssh command to start node in desired cluster via SLURM
+        withCredentials([
+            usernamePassword(
+                credentialsId: 'svc_tensorrt',
+                usernameVariable: 'USERNAME',
+                passwordVariable: 'PASSWORD'
+            )
+        ]) {
+            def remote = [
+                    ip           : cluster.ip,
+                    host         : cluster.host,
+                    user         : "${pipeline.USERNAME}",
+                    passwd       : "${pipeline.PASSWORD}",
+                    password     : "${pipeline.PASSWORD}",
+                    allowAnyHosts: true,
+            ]
+            Utils.exec(pipeline, script: "apt-get update && apt-get install -y sshpass openssh-client")
+            def jobWorkspace = "/home/svc_tensorrt/bloom/scripts/${jobUID}"
+            def tarName = BUILD_CONFIGS[config][TARNAME]
+            def llmTarfile = "https://urm.nvidia.com/artifactory/${ARTIFACT_PATH}/${tarName}"
+            def resourcePathNode = "/tmp/"
+            def llmSrcNode = "${resourcePathNode}TensorRT-LLM/src"
+            def isAarch64 = config.contains("aarch64")
+            def pytestTestTimeout = "7200"
+
+            stage('Prepare Testing') {
+                Utils.exec(pipeline, script: "sshpass -p '${remote.passwd}' ssh -oStrictHostKeyChecking=no ${remote.user}@${remote.host} 'mkdir ${jobWorkspace}'",)
+                def makoArgs = getMakoArgsFromStageName(stageName, true)
+                // TODO: currently the options will only be processed if the first
+                // line is "Mako options:", maybe we can make it more generic, which
+                // if the line cannot be split by "=", just ignore that line.
+                def makoOptsJson = transformMakoArgsToJson(["Mako options:"] + makoArgs)
+                def coverageConfigFile = "${jobWorkspace}/.coveragerc"
+                // generate coverage rc
+                coverageConfigDestPath = Utils.createTempLocation(pipeline, "./.coveragerc")
+                pipeline.writeFile(file: coverageConfigDestPath, text: """[run]
+branch = True
+data_file = ${jobWorkspace}/.coverage.${stageName}
+[paths]
+source =
+    ${llmSrcNode}/tensorrt_llm/
+    ---wheel_path---/tensorrt_llm/
+                """)
+                Utils.exec(pipeline, script: "chmod +w ${coverageConfigDestPath}", returnStdout: true)
+                Utils.exec(pipeline, script: "sshpass -p '${remote.passwd}' scp -r -p -oStrictHostKeyChecking=no ${coverageConfigDestPath} ${remote.user}@${remote.host}:${coverageConfigFile}",)
+                // generate multi node run script
+                scriptRunDestPath = Utils.createTempLocation(pipeline, "./slurm_run.sh")
+                pipeline.writeFile(file: scriptRunDestPath, text: """#!/bin/bash
```

Comment:
shall we put this `slurm_run.sh` into a batch script in the repo other than this. 
There are many funny formats here. And we should be able enhance this file by dev themselves w/o changing the jenkins file?